src/DssSpell.t.sol

```diff
@@ -659,139 +666,72 @@ contract DssSpellTest is DSTest, DSMath {
-        assertEq(bal.balanceOf(address(this)), ilkAmt);
-        assertEq(vat.gem("BAL-A", address(this)), 0);
-
-        // Generate new DAI to force a liquidation
-        bal.approve(address(joinBALA), ilkAmt);
-        joinBALA.join(address(this), ilkAmt);
-        (,,uint256 spotV,,) = vat.ilks("BAL-A");
-        // dart max amount of DAI
-        vat.frob("BAL-A", address(this), address(this), address(this), int(ilkAmt), int(mul(ilkAmt, spotV) / RAY));
-        hevm.warp(now + 1);
-        jug.drip("BAL-A");
-        assertEq(flipBALA.kicks(), 0);
-        cat.bite("BAL-A", address(this));
-        assertEq(flipBALA.kicks(), 1);
-    }
-
-    function testSpellIsCast_YFI_INTEGRATION() public {
-        vote();
-        scheduleWaitAndCast();
-        assertTrue(spell.done());
-
-        pipYFI.poke();
-        hevm.warp(now + 3601);
-        pipYFI.poke();
-        spot.poke("YFI-A");
-
-        // Add balance to the test address
-        uint256 ilkAmt = 1 * THOUSAND * WAD;
         hevm.store(
-            address(yfi),
-            keccak256(abi.encode(address(this), uint256(0))),
+            address(gusd_store),
+            keccak256(abi.encode(address(this), uint256(6))),
             bytes32(ilkAmt)
         );
-        assertEq(yfi.balanceOf(address(this)), ilkAmt);
+        assertEq(gusd.balanceOf(address(this)), ilkAmt);
 
         // Check median matches pip.src()
-        assertEq(pipYFI.src(), address(medYFIA));
+        //assertEq(pipGUSD.src(), address(medGUSDA));
 
         // Authorization
-        assertEq(joinYFIA.wards(pauseProxy), 1);
-        assertEq(vat.wards(address(joinYFIA)), 1);
-        assertEq(flipYFIA.wards(address(end)), 1);
-        assertEq(flipYFIA.wards(address(flipMom)), 1);
-        assertEq(pipYFI.wards(address(osmMom)), 1);
-        assertEq(pipYFI.bud(address(spot)), 1);
-        assertEq(pipYFI.bud(address(end)), 1);
-        assertEq(MedianAbstract(pipYFI.src()).bud(address(pipYFI)), 1);
+        assertEq(joinGUSDA.wards(pauseProxy), 1);
+        assertEq(vat.wards(address(joinGUSDA)), 1);
+        assertEq(flipGUSDA.wards(address(end)), 1);
+        assertEq(flipGUSDA.wards(address(flipMom)), 1);
+        //assertEq(pipGUSD.wards(address(osmMom)), 1);
+        //assertEq(pipGUSD.bud(address(spot)), 1);
+        //assertEq(pipGUSD.bud(address(end)), 1);
+        //assertEq(MedianAbstract(pipGUSD.src()).bud(address(pipGUSD)), 1);
 
         // Join to adapter
-        assertEq(vat.gem("YFI-A", address(this)), 0);
-        yfi.approve(address(joinYFIA), ilkAmt);
-        joinYFIA.join(address(this), ilkAmt);
-        assertEq(yfi.balanceOf(address(this)), 0);
-        assertEq(vat.gem("YFI-A", address(this)), ilkAmt);
+        assertEq(vat.gem("GUSD-A", address(this)), 0);
+        gusd.approve(address(joinGUSDA), ilkAmt);
+        joinGUSDA.join(address(this), ilkAmt);
+        assertEq(gusd.balanceOf(address(this)), 0);
+        assertEq(vat.gem("GUSD-A", address(this)), ilkAmt * 10**16);
```

Comment:
I'd suggest making an extra variable ilkAmt18 and using that one. So in collaterals that are 18 decimals we will just do `ilkAmt18 = ilkAmt` and we do not need to change several lines of the tests.

Reply:
I can update this, but we should probably spend some time making the integration test more reusable.

Reply:
Yeah let's leave this for another moment.